crate/server/src/core/cover_crypt/update_policy.rs

```diff
@@ -265,3 +220,70 @@ async fn refresh_all_user_decryption_keys(
 
     Ok(())
 }
+
+/// Update a Covercrypt policy based on the specified action and master private key.
+///
+/// This function takes an `EditPolicyAction` and a master private key as input and performs the
+/// corresponding policy updates. It retrieves the current policy from the master private key's
+/// attributes and applies the requested action, resulting in a new policy and a list of attributes
+/// that were updated or modified.
+///
+/// # Parameters
+///
+/// - `action`: An `EditPolicyAction` enum representing the action to perform on the policy.
+/// - `master_private_key`: A reference to the master private key object from which the policy
+///   attributes are retrieved.
+///
+/// # Returns
+///
+/// Returns a `Result` containing a tuple with the updated `Policy` and a vector of `Attribute`s
+/// that were modified during the action. If any errors occur during the policy updates or
+/// attribute retrieval, an `Err` variant is returned with a `KmsError`.
+fn dispatch_rekey_action(
+    action: EditPolicyAction,
+    master_private_key: &Object,
+) -> KResult<(Policy, Vec<Attribute>)> {
+    // Recover the current policy
+    let private_key_attributes = master_private_key.attributes()?;
+    let mut policy = policy_from_attributes(private_key_attributes)?;
+
+    match &action {
+        EditPolicyAction::RotateAttributes(attrs) => {
+            attrs.iter().try_for_each(|attr| policy.rotate(attr))
+        }
+        EditPolicyAction::ClearOldRotations(attrs) => attrs
+            .iter()
+            .try_for_each(|attr| policy.clear_old_attribute_values(attr)),
+        EditPolicyAction::RemoveAttribute(attrs) => attrs
+            .iter()
+            .try_for_each(|attr| policy.remove_attribute(attr)), // TODO: revoke existing keys with deleted attribute?)
+        EditPolicyAction::DisableAttribute(attrs) => attrs
+            .iter()
+            .try_for_each(|attr| policy.disable_attribute(attr)),
+        EditPolicyAction::RenameAttribute(pairs_attr_name) => pairs_attr_name
+            .iter()
+            .try_for_each(|(attr, new_name)| policy.rename_attribute(attr, new_name)), // TODO: rename attributes in existing AccessPolicy
+        EditPolicyAction::AddAttribute(attrs_properties) => {
+            attrs_properties
+                .iter()
+                .try_for_each(|(attr, encryption_hint)| {
+                    policy.add_attribute(attr.clone(), *encryption_hint)
+                })
+        }
+    }
+    .map_err(|e| {
+        KmsError::KmipError(
+            ErrorReason::Unsupported_Cryptographic_Parameters,
+            e.to_string(),
+        )
+    })?;
+
+    let attributes_to_update = match action {
+        EditPolicyAction::RotateAttributes(attrs) => attrs,
+        EditPolicyAction::ClearOldRotations(attrs) => attrs,
+        _ => vec![], // no need to update existing usk
+    };
```

Comment:
Several points:
1. the naming makes it unclear whether some more action needs to be performed on these attributes;
2. the naming is not coherent with the function doc which clearly states that an action has been performed on these attributes;
3. you may not need this second match (which would avoid a clone in the `EditPolicyAction::AddAttribute` case.